### PR TITLE
10 get api reviews queries

### PR DESCRIPTION
**Pull request by JJJUhlar**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -71,14 +70,53 @@ describe('appTests', () => {
...
+            .get('/api/reviews')
+            .expect(200)
+            .then(({body})=>{
+                expect(Array.isArray(body.reviews)).toBe(true)
+                expect(Object.prototype.toString.call(body.reviews[0])).toBe('[object Object]')
+                expect(Object.keys(body.reviews[0]).length).toBe(9)
+            })
+        })
+        
+
```
Comment (johnhutcheon) [inline __tests__/app.test.js:117]: Could we add another 200 test please for valid category query but has no reviews, which responds with an empty array of articles?

File: _models/models.js [hunk 2]
```diff
@@ -7,17 +8,40 @@ exports.selectCategories = () => {
...
+
+    let query = `
+                SELECT reviews.review_id, reviews.owner, reviews.title, reviews.category, reviews.review_img_url, reviews.created_at, reviews.votes, reviews.designer,  COUNT(comments.body) AS comment_count
+                FROM comments
+                RIGHT JOIN reviews
+                ON comments.review_id = reviews.review_id
+                WHERE category = $1
+                GROUP BY reviews.review_id
+                ORDER BY reviews.${sort_by} ${order};
+                `;
```
Comment (johnhutcheon) [inline _models/models.js:35]: We should use $2 and $3 instead of sort_by and order here, to avoid SQL injection if these are invalid.

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hey Joseph, could you just check my comments and make the amendments please? Thanks!

File: _models/models.js (hunk 2 above)
Comment (yalicee) [reply _models/models.js:35]: Please use what John mentioend above

**Review by yalicee — CHANGES_REQUESTED** [review]
to keep the code DRY, the query string should be built up dynamically according to the presence of the query. ie there shouldn't be multiple complete query strings within the code with one being conditionally selected according to the queries. Also see John's comment about SQL injection!!!
